caffe2/CMakeLists.txt

```diff
@@ -1236,7 +1189,7 @@ endif()
 if(USE_DISTRIBUTED)
   # Needed to support the inclusion of c10d/Foo.hpp headers.
   target_include_directories(torch_cpu PUBLIC ${TORCH_SRC_DIR}/lib)
-  target_compile_definitions(torch_cpu PRIVATE USE_DISTRIBUTED)
```

Comment:
Making this definition (and the ones below) public so that they are inherited by libtorch_python and by all the tests and other targets that depend on this and include its headers.